src/app/components/trip-details/trip-details.component.ts

```diff
@@ -69,6 +73,11 @@ export class TripDetailsComponent {
     });
   }
 
+  getDateWithoutTime(date: Date): Date {
```

Comment:
I see that you use this function a lot in the template, i would suggest removing all the logic from here and do it all in the selector, so all the UI does is just display

Reply:
I think you are right. I can do that 👍🏻